### PR TITLE
Operations index and improvments

### PR DESCRIPTION
**Pull request by miiu96**
- Added an extra Elasticsearch index `operations` where will be saved the  `transactions` and the `smart contract results`
- Removed the `pending-balances` for `accountsesdt` index

### CODE REVIEW

File: integrationtests/esdtTransfer_test.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+//go:build integration
```
Comment (bogdan-rosianu) [inline integrationtests/esdtTransfer_test.go:1]: might rename to `integrationtests` as `integration` could sound like a production build or something similar

File: process/elasticProcessor_test.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"github.com/ElrondNetwork/elastic-indexer-go/process/operations"
```
Comment (bogdan-rosianu) [inline process/elasticProcessor_test.go:9]: sort imports

File: process/interface.go [hunk 3]
```diff
@@ -102,3 +102,9 @@ type DBLogsAndEventsHandler interface {
 	SerializeTokens(tokens []*data.TokenInfo) ([]*bytes.Buffer, error)
 	SerializeDelegators(delegators map[string]*data.Delegator) ([]*bytes.Buffer, error)
 }
+
+// OperationsHandler defines the actions that an operations' handler should do
+type OperationsHandler interface {
+	ProcessTransactionsAndSCRS(txs []*data.Transaction, scrs []*data.ScResult) ([]*data.Transaction, []*data.ScResult)
```
Comment (bogdan-rosianu) [inline process/interface.go:108]: `SCRs` suffix to both functions?

File: process/logsevents/fungibleESDTProcessor.go [hunk 4]
```diff
@@ -82,33 +82,29 @@ func (fep *fungibleESDTProcessor) processEventOnSenderShard(event coreData.Event
 	})
 }
 
-func (fep *fungibleESDTProcessor) processEventDestination(args *argsProcessEvent, senderShardID uint32, selfShardID uint32) (string, string) {
+func (fep *fungibleESDTProcessor) processEventDestination(
+	args *argsProcessEvent,
+	selfShardID uint32,
+) (string, string, string, uint32) {
```
Comment (bogdan-rosianu) [inline process/logsevents/fungibleESDTProcessor.go:88]: could have used a struct instead of returning 4 values

File: process/logsevents/interface.go [hunk 5]
```diff
@@ -19,11 +18,13 @@ type argsProcessEvent struct {
...
 type argOutputProcessEvent struct {
-	identifier string
-	value      string
-	tokenInfo  *data.TokenInfo
-	delegator  *data.Delegator
-	processed  bool
+	identifier      string
+	value           string
+	receiver        string
+	receiverShardID uint32
```
Comment (bogdan-rosianu) [inline process/logsevents/interface.go:24]: place `receiverShardID` before `processed` -> mem alignment

File: process/operations/operationsProcessor.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+		return nil, indexer.ErrNilShardCoordinator
+	}
+
+	return &operationsProcessor{
+		shardCoordinator: shardCoordinator,
+		importDBMode:     importDBMode,
+	}, nil
+}
+
+func (op *operationsProcessor) ProcessTransactionsAndSCRS(
```
Comment (bogdan-rosianu) [inline process/operations/operationsProcessor.go:27]: add comment

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: process/miniblocks/miniblocksProcessor.go [hunk 7]
```diff
@@ -18,10 +18,10 @@ import (
 var log = logger.GetOrCreate("indexer/process/miniblocks")
 
 type miniblocksProcessor struct {
-	importDBMode bool
 	hasher       hashing.Hasher
 	marshalier   marshal.Marshalizer
 	selfShardID  uint32
+	importDBMode bool
```
Comment (dragos-rebegea) [inline process/miniblocks/miniblocksProcessor.go:24]: Is there any reason for the order change? If not, i won`t make unnecessary changes.

File: process/operations/operationsProcessor.go [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+// NewOperationsProcessor will create a new instance of operationsProcessor
+func NewOperationsProcessor(importDBMode bool, shardCoordinator indexer.ShardCoordinator) (*operationsProcessor, error) {
+	if check.IfNil(shardCoordinator) {
+		return nil, indexer.ErrNilShardCoordinator
+	}
+
+	return &operationsProcessor{
+		shardCoordinator: shardCoordinator,
```
Comment (dragos-rebegea) [inline process/operations/operationsProcessor.go:22]: maybe keep the same order as the one declared

File: process/operations/serialize.go [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+	scr *data.ScResult,
+) ([]byte, []byte, error) {
+	metaData := []byte(fmt.Sprintf(`{"update":{"_id":"%s", "_type": "_doc"}}%s`, scr.Hash, "\n"))
+	marshaledSCR, err := json.Marshal(scr)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	selfShardID := op.shardCoordinator.SelfId()
+	isCrossShardOnSourceShard := scr.SenderShard != scr.ReceiverShard && scr.SenderShard == selfShardID
```
Comment (dragos-rebegea) [inline process/operations/serialize.go:40]: i don\'t think `&& scr.SenderShard == selfShardID` is necessary

File: integrationtests/esdtTransfer_test.go (hunk 1 above)
Comment (miiu96) [reply integrationtests/esdtTransfer_test.go:1]: done.

File: process/elasticProcessor_test.go (hunk 2 above)
Comment (miiu96) [reply process/elasticProcessor_test.go:9]: done.

File: process/interface.go (hunk 3 above)
Comment (miiu96) [reply process/interface.go:108]: done.

File: process/logsevents/fungibleESDTProcessor.go (hunk 4 above)
Comment (miiu96) [reply process/logsevents/fungibleESDTProcessor.go:88]: this function will be refactored in the next PR

File: process/logsevents/interface.go (hunk 5 above)
Comment (miiu96) [reply process/logsevents/interface.go:24]: this will be refactored in the next PR

File: process/operations/operationsProcessor.go (hunk 6 above)
Comment (miiu96) [reply process/operations/operationsProcessor.go:27]: added

File: process/miniblocks/miniblocksProcessor.go (hunk 7 above)
Comment (miiu96) [reply process/miniblocks/miniblocksProcessor.go:24]: for memory aligment

File: process/operations/operationsProcessor.go (hunk 8 above)
Comment (miiu96) [reply process/operations/operationsProcessor.go:22]: done.

File: process/operations/serialize.go (hunk 9 above)
Comment (miiu96) [reply process/operations/serialize.go:40]: I want to know also if we are on the source shard

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]
